.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+# pre-commit is a tool to perform a predefined set of tasks manually and/or
+# automatically before git commits are made.
+#
+# Config reference: https://pre-commit.com/#pre-commit-configyaml---top-level
+#
+# Common tasks
+#
+# - Run on all files:   pre-commit run --all-files
+# - Register git hooks: pre-commit install --install-hooks
+#
+repos:
+  # Autoformat: Python code
+  - repo: https://github.com/ambv/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: [--target-version=py36]
+
+  # Autoformat: markdown, yaml
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.2.1
+    hooks:
+      - id: prettier
+        exclude: COPYING.md
+
+  # Lint: Python code
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: "3.8.4"
+    hooks:
+      - id: flake8
```

Comment:
I like the combination of black/prettier/flake8.
@minrk suggested some others in https://github.com/jupyterhub/kubespawner/pull/473/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9 (copied from jupyterhub), do you think it's worth adding those too and trying to standardise this across all Python repos, or are we better off leaving it to each repo?

Reply:
> do you think it's worth adding those too and trying to standardise this across all Python repos, or are we better off leaving it to each repo?

Can we delegate this to a team-compass issue? I'd like to stay clear of more work to consider those as part of this PR, but I know that prettier have done a great job with json / yaml for example, and even within markdown files that has code blocks of json / yaml - so various of those pre-commit hooks become irrelevant I think.

Very positive to prettier as an autoformatter with the piloting epxerience in z2jh btw!

Reply:
I went ahead and added some misc content anyhow in b508b75

check-yaml / check-json is covered by prettier, and flake8 is deprecated and we are using the new dedicated pre-commit hook in the flake8 official repo.